sound/soc/sof/intel/hda.c

```diff
@@ -79,10 +78,26 @@ static u32 hda_get_interface_mask(struct snd_sof_dev *sdev)
 	default:
 		break;
 	}
+}
+
+static u32 hda_get_interface_mask(struct snd_sof_dev *sdev)
+{
+	u32 interface_mask[2] = { 0 };
+
+	hda_get_interfaces(sdev, interface_mask);
 
 	return interface_mask[sdev->dspless_mode_selected];
 }
 
+bool hda_is_dai_supported(struct snd_sof_dev *sdev, u32 dai_type, bool dspless_mode)
+{
+	u32 interface_mask[2] = { 0 };
+
+	hda_get_interfaces(sdev, interface_mask);
+
+	return interface_mask[dspless_mode] & BIT(dai_type);
```

Comment:
I would be tempted to
```c
return !!(interface_mask[dspless_mode] & BIT(dai_type));
```

Reply:
not sure why? Leaving as is for now.

Reply:
obsolete, moved to a non-ambiguous 'is_chain_dma_supported' callback